### PR TITLE
Added JugglerSequence.js

### PR DESCRIPTION
**Pull request by tehliang**
[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

- [x] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### Checklist:

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new JavaScript files are placed inside an existing directory.
- [x] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: Maths/JugglerSequence.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  // Calculate terms until last term is not 1
+  while (a !== 1) {
+    let b = 0
+    // Check if previous term is even or odd
+    if (a % 2 === 0) {
+    // calculate next term
+      b = Math.floor(Math.sqrt(a))
+    } else {
+      // for odd previous term calculate next term
+      b = Math.floor(Math.sqrt(a) * Math.sqrt(a) * Math.sqrt(a))
```
Comment (appgurueu) [inline Maths/JugglerSequence.js:26]: `sqrt(a)^3 = sqrt(a)^2 * sqrt(a) = a * sqrt(a) = a ^ 1.5` (`a ** 1.5` in JS), please simplify; once you have simplified this, you might simplify the entire `if-else` to a ternary:

`b = (a % 2 === 0) ? Math.sqrt(a) : (a * Math.sqrt(a))`

which you then may simplify to

`b = ((a % 2 === 0) ? 1 : a) * Math.sqrt(a)`

which can alternatively be written as

`b = a ** (a % 2 === 0) ? 0.5 : 1.5)`

which can again be simplified to

`b = a ** ((a % 2) + 0.5)`

File: Maths/JugglerSequence.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ * isCompositeNumber(15) // returns [15, 58, 7, 18, 4, 2, 1]
+ */
+
+function jugglerSequence (n) {
+  const sequence = []
+  let a = n
+  sequence.push(n)
+  // Calculate terms until last term is not 1
+  while (a !== 1) {
+    let b = 0
```
Comment (appgurueu) [inline Maths/JugglerSequence.js:19]: Why would you need a new variable here - can't you just write to `a`?

File: Maths/test/JugglerSequence.test.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { jugglerSequence } from '../JugglerSequence'
+
+describe('Testing isCompositeNumber function', () => {
```
Comment (appgurueu) [inline Maths/test/JugglerSequence.test.js:3]: Copy-paste remnants: Reads "isCompositeNumber"

File: Maths/JugglerSequence.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+/*
+ * Juggler Sequence: https://en.wikipedia.org/wiki/Juggler_sequence
+ * function jugglerSequence
+ * Juggler Sequence is a series of integer number in which the first term starts with a positive integer number n
+ * and the remaining terms are generated from the immediate previous term using the recurrence relation
+ * Produce Juggler Sequence using number n as the first term of the sequence and store in an array
+ * Reference: https://www.geeksforgeeks.org/juggler-sequence/
+ * jugglerSequence(3) // returns [3, 5, 11, 36, 6, 2, 1 ]
+ * isCompositeNumber(9) // returns [9, 27, 140, 11, 36, 6, 2, 1]
```
Comment (appgurueu) [inline Maths/JugglerSequence.js:9]: Again copy-paste remnants

File: Maths/JugglerSequence.js (hunk 1 above)
Comment (tehliang) [reply Maths/JugglerSequence.js]: Still need Math.floor() function to return the largest integer less than or equal to a given number.
Thanks a lot!

File: Maths/test/JugglerSequence.test.js (hunk 3 above)
Comment (tehliang) [reply Maths/test/JugglerSequence.test.js]: Hmm, my bad. Should just draft pull request first.

File: Maths/JugglerSequence.js (hunk 4 above)
Comment (tehliang) [reply Maths/JugglerSequence.js]: Hmm my bad. Should just draft pull request first.

**Review by appgurueu — APPROVED** [review]

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**